sdk/core/Azure.Core/src/Serialization/AzureJsonSerializer.cs

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.IO;
+using System.Reflection;
+using System.Text.Json;
+using System.Threading.Tasks;
+
+#pragma warning disable CS1591, SA1649, SA1402
+
+namespace Azure.Core.Serialization
+{
+    // For libraries that operate on opaque binary data (event hubs)
+    public abstract class AzureObjectSerializerFactory
+    {
+        public abstract AzureObjectSerializer Create(Type type);
+        // Async to allow to support scenarios where schema is retrieved from somewhere
+        public abstract ValueTask<AzureObjectSerializer> CreateAsync(Type type);
+    }
+
+    // For libraries that need to have json specific features (tables, search)
+    public abstract class AzureJsonSerializerFactory: AzureObjectSerializerFactory
+    {
+        public abstract AzureJsonSerializer CreateJsonSerializer(Type type);
+        public abstract ValueTask<AzureObjectSerializer> CreateJsonSerializerAsync(Type type);
+    }
+
+    // Represents a serializer/deserializer from any byte stream to an object
+    public abstract class AzureObjectSerializer
```

Comment:
This does not work for AVRO. AVRo needs support for streaming records, i.e. something like enumerating objects from a stream.

Reply:
Can't you keep calling the deserialize on the same stream multiple times?

Reply:
Am not familiar with AVRO but one Q:
Is the stream expected to be processed with the expectation of possible future fault/error? 

ex: Assuming stream is for list, if later parts of the entries are illformatted/corrupted.

Reply:
We've decided that this abstraction would only be used to deserialize singular instances and we'll build another for streaming (AVRO) when the need  arises